go/test/endtoend/vreplication/vreplication_vtctldclient_cli_test.go

```diff
@@ -184,6 +193,135 @@ func createMoveTables(t *testing.T, sourceKeyspace, targetKeyspace, workflowName
+
+	rs.ReverseReadsAndWrites()
+	waitForLowLag(t, keyspace, workflowName)
+	vdiff(t, keyspace, workflowName, "zone1", false, true, nil)
+	rs.SwitchReadsAndWrites()
+	rs.Complete()
+}
+
+func getReshardShowResponse(rs *iReshard) *vtctldatapb.GetWorkflowsResponse {
+	(*rs).Show()
+	reshardOutput := (*rs).GetLastOutput()
+	var reshardResponse vtctldatapb.GetWorkflowsResponse
+	err := protojson.Unmarshal([]byte(reshardOutput), &reshardResponse)
+	require.NoError(vc.t, err)
+	reshardResponse.Workflows[0].MaxVReplicationTransactionLag = 0
+	reshardResponse.Workflows[0].MaxVReplicationLag = 0
+	return reshardResponse.CloneVT()
+}
+
+func validateReshardResponse(rs iReshard) {
+	resp := getReshardResponse(rs)
+	require.NotNil(vc.t, resp)
+	require.NotNil(vc.t, resp.ShardStreams)
+	require.Equal(vc.t, len(resp.ShardStreams), 2)
+	keyspace := "customer"
+	for _, shard := range []string{"-40", "40-80"} {
+		streams := resp.ShardStreams[fmt.Sprintf("%s/%s", keyspace, shard)]
+		require.Equal(vc.t, 1, len(streams.Streams))
+		require.Equal(vc.t, binlogdatapb.VReplicationWorkflowState_Stopped.String(), streams.Streams[0].Status)
+	}
+}
+
+func validateReshardWorkflow(t *testing.T, workflows []*vtctldatapb.Workflow) {
+	require.Equal(t, 1, len(workflows))
+	wf := workflows[0]
+	require.Equal(t, "reshard", wf.Name)
+	require.Equal(t, binlogdatapb.VReplicationWorkflowType_Reshard.String(), wf.WorkflowType)
+	require.Equal(t, "None", wf.WorkflowSubType)
+	require.Equal(t, "customer", wf.Target.Keyspace)
+	require.Equal(t, 2, len(wf.Target.Shards))
+	require.Equal(t, "customer", wf.Source.Keyspace)
+	require.Equal(t, 1, len(wf.Source.Shards))
+	require.False(t, wf.DeferSecondaryKeys)
+
+	require.GreaterOrEqual(t, len(wf.ShardStreams), int(1))
+	oneStream := maps.Values(wf.ShardStreams)[0]
+	require.NotNil(t, oneStream)
+
+	stream := oneStream.Streams[0]
+	require.Equal(t, binlogdatapb.VReplicationWorkflowState_Stopped.String(), stream.State)
+	require.Equal(t, stream.TabletSelectionPreference, tabletmanagerdatapb.TabletSelectionPreference_INORDER)
+	require.True(t, slices.Equal([]topodatapb.TabletType{topodatapb.TabletType_PRIMARY, topodatapb.TabletType_RDONLY}, stream.TabletTypes))
+	require.True(t, slices.Equal([]string{"zone1", "zone2"}, stream.Cells))
+
+	bls := stream.BinlogSource
+	require.Equal(t, binlogdatapb.OnDDLAction_STOP, bls.OnDdl)
+	require.True(t, bls.StopAfterCopy)
+
+}
+
+func getReshardResponse(rs iReshard) *vtctldatapb.WorkflowStatusResponse {
+	reshardOutput := rs.GetLastOutput()
+	var reshardResponse vtctldatapb.WorkflowStatusResponse
+	err := protojson.Unmarshal([]byte(reshardOutput), &reshardResponse)
+	require.NoError(vc.t, err)
+	return reshardResponse.CloneVT()
+}
+
+// helper functions
+
+func getWorkflow(targetKeyspace, workflow string) *vtctldatapb.GetWorkflowsResponse {
+	workflowOutput, err := vc.VtctldClient.ExecuteCommandWithOutput("Workflow", "--keyspace", targetKeyspace, "show", "--workflow", workflow)
```

Comment:
FWIW, it's more compact if we also pass `--compact --include-logs=false`. Then we can also test those two flags.